sdk/core/Azure.Core/src/Pipeline/BearerTokenAuthenticationPolicy.cs

```diff
@@ -13,35 +15,28 @@ namespace Azure.Core.Pipeline
     /// </summary>
     public class BearerTokenAuthenticationPolicy : HttpPipelinePolicy
     {
-        private readonly TokenCredential _credential;
-
-        private readonly string[] _scopes;
-
-        private string? _headerValue;
-
-        private DateTimeOffset _refreshOn;
+        private readonly AccessTokenCache _accessTokenCache;
 
         /// <summary>
         /// Creates a new instance of <see cref="BearerTokenAuthenticationPolicy"/> using provided token credential and scope to authenticate for.
         /// </summary>
         /// <param name="credential">The token credential to use for authentication.</param>
         /// <param name="scope">The scope to authenticate for.</param>
-        public BearerTokenAuthenticationPolicy(TokenCredential credential, string scope) : this(credential, new[] { scope })
-        {
-        }
+        public BearerTokenAuthenticationPolicy(TokenCredential credential, string scope) : this(credential, new[] { scope }) { }
 
         /// <summary>
         /// Creates a new instance of <see cref="BearerTokenAuthenticationPolicy"/> using provided token credential and scopes to authenticate for.
         /// </summary>
         /// <param name="credential">The token credential to use for authentication.</param>
         /// <param name="scopes">Scopes to authenticate for.</param>
         public BearerTokenAuthenticationPolicy(TokenCredential credential, IEnumerable<string> scopes)
-        {
+            : this(credential, scopes, TimeSpan.FromMinutes(5), TimeSpan.FromSeconds(30)) { }
+
+        internal BearerTokenAuthenticationPolicy(TokenCredential credential, IEnumerable<string> scopes, TimeSpan tokenRefreshOffset, TimeSpan tokenRefreshRetryTimeout) {
```

Comment:
Can we rename `tokenRefreshRetryTImeout` everywhere? I know it's all internal, but I think the name doesn't really express what it is used for and so it's more difficult to undertand the code. I think `retryDelay` or `retryRate` would be more discriptive. It would need to be changed bot there and in `AccessTokenCache`.

Reply:
Fixed